art/estimators/classification/blackbox.py

```diff
@@ -287,3 +306,91 @@ def loss(self, x: np.ndarray, y: np.ndarray, **kwargs) -> np.ndarray:
 
     def compute_loss(self, x: np.ndarray, y: np.ndarray, **kwargs) -> np.ndarray:
         raise NotImplementedError
+
+
+@total_ordering
+class FuzzyMappingWrapper:
+    """
+    Wrapper class for a sample, label pair to be used in a `SortedList`.
+    """
+
+    def __init__(self, key: np.ndarray, value=None):
+        """
+        Create a wrapper for the key, value to pair to be used in a `SortedList`.
+
+        :param key: The sample to be matched against.
+        :param value: The mapped value.
+        """
+        self.key = key
+        self.value = value
+
+    def __eq__(self, other):
+        return np.all(np.isclose(self.key, other.key))
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __ge__(self, other):
```

Comment:
I haven't used `@total_ordering` before and looked it up in the docs and noticed the note that it comes at the cost of slower execution. You have done performance tests with `SortedList`, does `SortedList` only use `__ge__` or could explicitly defining all comparison magic functions further improve performance here?

Reply:
It seems `SortedList` uses python's `bisect`, which does use all comparison types. I ran the tests using `@total_ordering` and without, and the performance difference seems negligible (on the order of ms). But I did notice that by not calling `__eq__` in `__ge__` a repeated call to `np.isclose` can be removed, so I was able to halve the running time.

Reply:
That's great news!